include/deal.II/base/exceptions.h

```diff
@@ -1278,8 +1278,12 @@ namespace internal
  * @ingroup Exceptions
  * @author Bruno Turcksin, 2016
  */
+#ifdef DEBUG
 #define AssertCuda(error_code) Assert(error_code == cudaSuccess, \
                                       dealii::ExcCudaError(cudaGetErrorString(error_code)))
+#else
+#define AssertCuda(error_code) { (void) error_code; }
```

Comment:
Since `error_code` can be a complicated expression, you probably want to write this as
```
  (void) (error_code);
```
But the substitution is not the same -- if `error_code` involves function calls, the old code omitted these function calls in release mode, whereas now you are actually evaluating them. I don't know whether that happens anywhere, but this is the reason why `Assert` does not evaluate its operand in release mode.

Reply:
`error_code` is always an `int`, it cannot be anything else but I can change the code if you want.

Reply:
But it's conceivable that you call the macro as in `AssertCuda(some_object.check_internal_consistency())` where the inner function call returns an `int`.

Reply:
I can't think of a case where that would make sense to do that but I'll change the code just in case.

Reply:
I'm sure I could find a number of places in the library where we call expensive functions inside an `Assert`. It may just be the style in the CUDA code to never do that and to always only use the macro to check return codes -- that's what we do for the PETSc wrappers, for example, and also for MPI. If that's the case, that's ok with me then, but it's worth remembering that one can, in principle, call an expensive function inside an assertion and that that should not happen in release mode.